src/lib/offchain.ts

```diff
@@ -94,6 +109,79 @@ async function getUsdcTokenData(
   return attestations
 }
 
+/**
+ * Returns the LBTC attestation for a given payload hash
+ *
+ * @param payloadHash - hash of the payload of the LBTC transfer
+ * @param isTestnet - true if this was from a testnet
+ * @returns LBTC attestation bytes
+ */
+async function getLbtcAttestation(payloadHash: string, isTestnet: boolean): Promise<string> {
+  const lbtcApiBaseUrl = isTestnet ? LOMBARD_API_URL.testnet : LOMBARD_API_URL.mainnet
+  const res = await fetch(`${lbtcApiBaseUrl}/bridge/v1/deposits/getByHash`, {
+    method: 'POST',
+    body: JSON.stringify({ messageHash: [payloadHash] }),
+  })
+  const response = (await res.json()) as LombardAttestationsResponse
+  if (response == null || !('attestations' in response)) {
+    throw new Error(
+      'Error while fetching LBTC attestation. Response: ' + JSON.stringify(response, null, 2),
+    )
+  }
+  const attestation = response.attestations.find((att) => att.message_hash === payloadHash)
+  if (attestation == null) {
+    throw new Error(
+      'Could not find requested LBTC attestation with hash:' +
+        payloadHash +
+        ' in response: ' +
+        JSON.stringify(response, null, 2),
+    )
```

Comment:
It's better to not throw inside this function, and instead return `undefined`, otherwise, a (malicious?) TokenPool emitting events similar to your LBTC's can DoS your manualExec algo. The USDC's approach above is: let's be optimist, do as many checks as you can/need, perform the request, but if anything fail, let it pass through undefined, and possibly (if it was a transient error) fail at estimation time (e.g. dest TP requires the attestation that we failed to fetch), but will eventually succeed if/when the user retries, if it should.

Reply:
done

Reply:
Is it? I still see a bunch of throws in this function